core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncFrequency.java

```diff
@@ -0,0 +1,168 @@
+     * If offline and the change was local, the application will queue the event to
+     * be synchronized when back online.
+     */
+    REACTIVE,
+    /**
+     * Local/remote events will be queued on the device for
+     * a specified amount of time (configurable) before they are applied.
+     */
+    SCHEDULED,
+    /**
+     * The collection will only sync changes when
+     * specified by the application.
+     */
+    ON_DEMAND
+  }
+
+  /**
+   * When a change is made locally or a change from a remote collection is received,
+   * the Stitch application will react to the event immediately.
+   * If offline and the change was local, the application will queue the event to
+   * be synchronized when back online.
+   */
+  static class ReactiveFrequency extends SyncFrequency {
+    @Override
+    SyncFrequencyType getType() {
+      return SyncFrequencyType.REACTIVE;
+    }
+  }
+
+  /**
+   * Local/remote events will be queued on the device for
+   * a specified amount of time (configurable) before they are applied.
+   */
+  static class ScheduledFrequency extends SyncFrequency {
+    private final long timeInterval;
+    private final TimeUnit timeUnit;
+    private final boolean isConnected;
+
+    ScheduledFrequency(final long timeInterval,
+                       final @Nonnull TimeUnit timeUnit,
+                       final boolean isConnected) {
+      this.timeInterval = timeInterval;
+      this.timeUnit = timeUnit;
+      this.isConnected = isConnected;
+    }
+
+    @Override
+    SyncFrequencyType getType() {
+      return SyncFrequencyType.SCHEDULED;
+    }
+
+    /**
+     * Whether or not the application continuously applying events by maintaining a sync stream
+     * @return true if continuously applying events; false if not
+     */
+    boolean isConnected() {
+      return isConnected;
+    }
+
+    /**
+     * The amount of time in the given {@link TimeUnit} between syncs.
+     * A thread will be scheduled to run at the associated interval.
+     * @return the interval
+     */
+    long getTimeInterval() {
+      return timeInterval;
+    }
+
+    /**
+     * The {@link TimeUnit} qualifier for the timeInterval number.
+     * @return the associated {@link TimeUnit}
+     */
+    TimeUnit getTimeUnit() {
+      return timeUnit;
+    }
+  }
+
+  /**
+   * The collection will only sync changes when
+   * specified by the application.
+   */
+  static class OnDemandFrequency extends SyncFrequency {
+    @Override
+    SyncFrequencyType getType() {
+      return SyncFrequencyType.ON_DEMAND;
+    }
+  }
+
+  /**
+   * A configuration to enforce that when a change is made locally or a change
+   * from a remote collection is received, the Stitch application
+   * will react to the event immediately. If offline and the change was local,
+   * the application will queue the event to be synchronized when back online.
+   * @return a reactive configuration
+   */
+  public static SyncFrequency newReactiveConfiguration() {
```

Comment:
For this kind of mechanic where it’s not a true factory pattern, I like fluent-like syntax better, eg SyncFrequency.reactive()

Also for reactive and on demand should be a singleton instance.